src/main/java/wooteco/subway/domain/Sections.java

```diff
@@ -0,0 +1,146 @@
+package wooteco.subway.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import wooteco.subway.exception.BothUpAndDownStationDoNotExistException;
+import wooteco.subway.exception.BothUpAndDownStationExistException;
+import wooteco.subway.exception.CanNotInsertSectionException;
+import wooteco.subway.exception.OnlyOneSectionException;
+
+public class Sections {
+
+    private final List<Section> sections;
+
+    public Sections(List<Section> sections) {
+        validateEmptySections(sections);
+        this.sections = new ArrayList<>(sections);
+    }
+
+    private void validateEmptySections(List<Section> sections) {
+        if (sections.isEmpty()) {
+            throw new IllegalArgumentException("비어있는 구간 목록은 생성할 수 없습니다.");
+        }
+    }
+
+    public void addSection(Section newSection) {
+        validateBothStationsExisting(newSection);
+        validateBothStationsNotExisting(newSection);
+
+        if (isNeededToInsert(newSection)) {
+            insertSection(newSection);
+        }
```

Comment:
왜 insert가 필요한지 코드에서 설명해줘야한다고 생각해요.
메서드 네이밍을 객체에게 메시지를 보낸다고, 생각하고 작성하면 좋을 것 같습니다.

```suggestion
        if (갈래길이 생길 수 있는가?(newSection)) {
            중간역수정(newSection);
        }
```